geniza/common/views.py

```diff
@@ -1,3 +1,26 @@
 from django.shortcuts import render
+from django.utils.translation import gettext as _
 
-# Create your views here.
+
+def error_404(request, *args, **kwargs):
+    """Render a 404 error with the correct status, context, and template."""
+    context = {
+        # Translators: title for Not Found (404) error page
+        "page_title": _("Not Found"),
+        # Translators: description for Not Found (404) error page
+        "page_description": _("Oops, you've hit a lacuna! Page not found."),
+        "page_type": "error",
+    }
+    return render(request, "404.html", context, status=404)
+
+
+def error_500(request):
+    """Render a 500 error with the correct status, context, and template."""
+    context = {
+        # Translators: title for Internal Server Error (500) error page
+        "page_title": _("Server Error"),
+        # Translators: description for Internal Server Error (500) error page
+        "page_description": _("Something went wrong putting this page together."),
+        "page_type": "error",
+    }
+    return render(request, "500.html", context, status=500)
```

Comment:
why are these needed? I don't see anything here that couldn't be done in the template directly

Reply:
I'll try again, but I believe the first time I tried without the views, the `site` context wasn't coming through via `context_processors`.

Reply:
@blms at least in the 500 error it's possible that it's intentionally not available — you don't want to trigger a new error when you're trying to render an error! we can check the django docs and look at other of our apps to see how we approached this; we need `site` to render menus/nav?

Reply:
Yes, that's why we need it. Good idea to check other projects!

Reply:
It breaks everywhere that `site.root_page` is used, which means:
1. Places where we loop through its children (nav menus in header and footer)
2. Places where we reference it directly to get its `pageurl` (logo in header, "Home" link in footer)

Reply:
from the documentation: https://docs.djangoproject.com/en/3.2/ref/views/#the-500-server-error-view 

> The default 500 view passes no variables to the 500.html template and is rendered with an empty Context to lessen the chance of additional errors.

Would it make sense to skip the nav on the 500 page to keep it as simple as possible? I think you should have access to normal context on the 400

Reply:
Done in ab7d30f 👍